### PR TITLE
Fix order of function parameters on @discord.ui.select

### LINKED ISSUE #1108: AttributeError: 'Select' object has no attribute 'data'
Sentry Issue: [SIR-LANCEBOT-8P](https://sentry.io/organizations/python-discord/issues/3635087462/?referrer=github_integration)

```
AttributeError: 'Select' object has no attribute 'data'
  File "discord/ui/view.py", line 425, in _scheduled_task
    await item.callback(interaction)
  File "bot/exts/utilities/epoch.py", line 124, in select_format
    selected = interaction.data["values"][0]

Ignoring exception in view "<TimestampMenuView timeout=60 children=1>" for item "<Select placeholder='Select the format of your timestamp' min_values=1 max_values=1 options=[<SelectOption label='Epoch' value='Epoch' description='1664568890' emoji=None default=False>, <SelectOption label='Short Time' value='Short Time' description='8:14 PM' emoji=None default=False>, <SelectOption label='Long Time' value='Long Time' description='8:14:50 PM' emoji=None default=False>, <SelectOption label='Short Date' value='Short Date' description='09/30/2022' emoji=None default=False>, <SelectOption l..."
```

### PR DESCRIPTION
**Pull request by TizzySaurus**
## Relevant Issues

Closes #1108.

## Description

During development of dpy 2.0, the order of the `interaction` and `select` swapped, but this wasn't updated in the codebase. This PR fixes that in the 3 use-cases of the `@discord.ui.select` decorator.

## Did you:

- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] Read all the comments in this template?
- [x] Ensure there is an issue open, or link relevant discord discussions?
- [x] Read and agree to the [contributing guidelines](https://pythondiscord.com/pages/contributing/contributing-guidelines/)?

### CODE REVIEW

**Review by HassanAbouelela — APPROVED** [review]
Looks good, thanks

File: bot/exts/events/advent_of_code/views/dayandstarview.py [hunk 1]
```diff
@@ -64,12 +64,12 @@ async def day_select(self, select: discord.ui.Select, interaction: discord.Inter
...
         custom_id="star_select"
     )
-    async def star_select(self, select: discord.ui.Select, interaction: discord.Interaction) -> None:
+    async def star_select(self, _: discord.Interaction, select: discord.ui.Select) -> None:
         """Dropdown to choose either the first or the second star."""
         self.star = select.values[0]
 
     @discord.ui.button(label="Fetch", style=discord.ButtonStyle.blurple)
-    async def fetch(self, button: discord.ui.Button, interaction: discord.Interaction) -> None:
+    async def fetch(self, _: discord.ui.Button, interaction: discord.Interaction) -> None:
```
Comment (wookie184) [inline bot/exts/events/advent_of_code/views/dayandstarview.py:72]: Is `Button` still meant to come before `Interaction`?

Comment (TizzySaurus) [reply bot/exts/events/advent_of_code/views/dayandstarview.py:72]: Good catch. This needs to be swapped, alongside the other uses of `@discord.ui.button`.

Comment (TizzySaurus) [reply bot/exts/events/advent_of_code/views/dayandstarview.py]: Fixed :+1:

File: bot/exts/events/trivianight/_scoreboard.py [hunk 2]
```diff
@@ -123,26 +123,26 @@ def _get_rank(self, member: Member) -> Embed:
...
         """
         Send an ephemeral message with the speed leaderboard embed.
 
         Parameters:
-            - button: The discord.ui.Button instance representing the `Speed Leaderboard` button.
-            - interaction: The discord.Interaction instance containing information on the interaction between the user
+             - interaction: The discord.Interaction instance containing information on the interaction between the user
             and the button.
+            - button: The discord.ui.Button instance representing the `Speed Leaderboard` button.
         """
```
Comment (wookie184) [inline bot/exts/events/trivianight/_scoreboard.py:134]: Extra space that shouldn't be there on the `- interaction:` line

Comment (TizzySaurus) [reply bot/exts/events/trivianight/_scoreboard.py]: Yep, fixed :+1:

**Review by wookie184 — APPROVED** [review]
Thanks!

**Pull request merged**